index.js

```diff
@@ -106,7 +110,15 @@ DB.prototype.configure = function (config) {
  * @return {Promise}
  */
 DB.prototype.query = function(query, params) {
-  return runQueryWith.call(this, query, params);
+  let i = 0;
+  query = query.replace(/\?/g, function (match){
+    let param = params[i++];
+    if (Array.isArray(param)){
+      return Array(param.length).fill('?').join(',')
+    }
+    return '?';
+  });
+  return runQueryWith.call(this, query, flattenDeep(params));
```

Comment:
the flattenDeep() is recursive will flatten the arrays in depth, but the loop here will go into depth 1 only, this could cause inconsistent data
eg: query = 'a IN (?)', params = [[[x, y, z]]] then the query will be: "a IN (?)" but the values will be ([x, y, z])

We have two options to solve this: 
* Validate the params to not be more than 2D array
* Flatten the params with the same way we add '?' (not recursive)

WDYT?